File: TravelHelperDesktopApp/src/DAO/DAOFactory.java
/*
 * Progetto di Ingegneria del Software A.A 2019-2020
 * CdL Informatica - Università di Napoli Federico II
 * Realizzato da Ernesto De Crecchio - N86001596
 */

package DAO;

/**
 *
 * @author ernestodecrecchio
 */
public class DAOFactory {
    
    public static StructureDAO getStructureDAO(String type) {
        if(type.equalsIgnoreCase("AWSElasticBeanstalk")) {
            return new StructureDAOAWSElasticBeanstalkImpl();
        } else { // Altre implementazioni possibili diverse da AWS Elastic Beanstalk
            return new StructureDAOAWSElasticBeanstalkImpl();
        }
    }
    
    public static ReviewDAO getReviewDAO(String type) {
        if(type.equalsIgnoreCase("AWSElasticBeanstalk")) {
            return new ReviewDAOAWSElasticBeanstalkImpl();
        } else { // Altre implementazioni possibili diverse da AWS Elastic Beanstalk
            return new ReviewDAOAWSElasticBeanstalkImpl();
        }
    }
    
    public static UserDAO getUserDAO(String type) {
        if(type.equalsIgnoreCase("AWSElasticBeanstalk")) {
            return new UserDAOAWSElasticBeanstalkImpl();
        } else { // Altre implementazioni possibili diverse da AWS Elastic Beanstalk
            return new UserDAOAWSElasticBeanstalkImpl();
        }
    }
    
    public static AccessDAO getAccessDAO(String type) {
        if(type.equalsIgnoreCase("AWSElasticBeanstalk")) {
            return new AccessDAOAWSElasticBeanstalkImpl();
        } else { // Altre implementazioni possibili diverse da AWS Elastic Beanstalk
            return new AccessDAOAWSElasticBeanstalkImpl();
        }
    }
}

File: TravelAppMobile/TravelAppMobile/DAO/DAOFactory.swift
//
//  DAOFactory.swift
//  TravelAppMobile
//
//  Created by Ernesto De Crecchio on 21/05/2020.
//  Copyright © 2020 Ernesto De Crecchio. All rights reserved.
//

import Foundation

public class DAOFactory {
    
    static func getStructureDAO(type: String) -> StructureDAOProtocol {
        if type == "AWSElasticBeanstalk" {
            return StructureDAOAWSElasticbeanstalkImpl()
        } else { // Altre implementazioni possibili diverse da AWS Elastic Beanstalk
            return StructureDAOAWSElasticbeanstalkImpl()
        }
    }
    
    static func getUserDAO(type: String) -> UserDAOProtocol {
        if type == "AWSElasticBeanstalk" {
            return UserDAOAWSElasticbeanstalkImpl()
        } else { // Altre implementazioni possibili diverse da AWS Elastic Beanstalk
            return UserDAOAWSElasticbeanstalkImpl()
        }
    }
    
    static func getReviewDAO(type: String) -> ReviewDAOProtocol {
        if type == "AWSElasticBeanstalk" {
            return ReviewDAOAWSElasticbeanstalkImpl()
        } else { // Altre implementazioni possibili diverse da AWS Elastic Beanstalk
            return ReviewDAOAWSElasticbeanstalkImpl()
        }
    }
    
    static func getAccessDAO(type: String) -> AccessDAOProtocol {
        if type == "AWSElasticBeanstalk" {
            return AccessDAOAWSElasticbeanstalkImpl()
        } else { // Altre implementazioni possibili diverse da AWS Elastic Beanstalk
            return AccessDAOAWSElasticbeanstalkImpl()
        }
    }
    
}
